security/WFCORE-5466_Remove_Legacy_Vault_Integration.adoc

```diff
@@ -0,0 +1,311 @@
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [x] QE
+
+As a removal RFE this does not require new feature test development, however both teams will need
+to review existing testsuites for tests dependent on the Vault integration.  Identified tests will
+either need deleting or converting to a credential store equivalent.
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+
+=== Other Interested Projects
+
+* [x] External Testsuites
+
+* [ ] Quickstarts
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+As a removal task, this section is describing the steps being undertaken rather than a set of
+"requirements".
+
+=== Hard Requirements
+
+The command line vault utility scripts `vault.bat`, `vault.ps1`, and `vault.sh` are all removed
+from the server.
+
+The next version of the management model is updated to remove all vault resources.  Any use of
+vault resources in the latest version will result in a usual unexpected element error.
+
+Where configurations are read using older model versions if they contain a vault definition this
+will be logged as a warning and the vault configuration dropped at the next write.  This is to
+allow older configurations to still be handled in admin mode.  This same approach will be used for
+appclient parsing.
+
+Vault expressions are not prevented from occurring within the management model for two reasons:
+
+* The WildFly Elytron expression encryption can use a similar format.
+* Domains will need to handle configurations containing vault expressions for older host controllers.
+
+The vault elements in the WildFly CLI config are removed from the next version of the schema.  The
+presence of the element in ALL versions of configuration will result in a parsing error being
+reported.  The parsing is more abrubt in the CLI as this is only used for configuring SSL so will
+need immediate attention.
+
+All vault integration code is deleted from WildFly Core reducing the dependence on PicketBox.
+
+All vault test utilities are deleted and removed from the testsuite.
+
+The `vault` galleon layer will be removed from the available galleon layers.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+This RFE is not quite sufficient to remove dependencies on PicketBox from WildFly Core.
+
+The legacy security subsystem also supports it's own vault definition, this is outside the
+scope of the task and will be handled with the removal of the subsystem.
+
+The core management vault resources are not transformed so configuration does not need to be
+retained for managing older host controllers.
+
+The following items identified during the removal of Vault integration will be handled
+independently in their own follow up tasks:
```

Comment:
WFLY-14919 and WFCORE-5490 look like RFEs. Do we want to treat those as RFEs or will those be part of another RFE? Do we need/want those in Alpha?

Reply:
I think both of those should be a follow up for later, at this point the priority is getting the old code out and then we can follow up.  We should probably clone both to JBEAPs for now with a target release of 8.0.0.GA so we have them in a single location and don't forget.